sentry-core/src/main/java/io/sentry/core/Scope.java

```diff
@@ -402,9 +413,30 @@ SessionPair startSession() {
       }
       previousSession = session;
 
-      session =
-          new Session(
-              options.getDistinctId(), user, options.getEnvironment(), options.getRelease());
+      String distinctId = null;
+
+      final User finalUser = getUser();
```

Comment:
Does this mean that now even if the user isn't cached we'll be user the user from the scope?
Seems like this brings us back to a potential double-user count per device issue.

Reply:
> Does this mean that now even if the user isn't cached we'll be user the user from the scope?

mm guess I don't get it, what do you mean?

> Seems like this brings us back to a potential double-user count per device issue.
 
this is true only on the first run of the App., if `cacheUserForSessions` is enabled and value has been set, it's gonna reuse the same distinctId, unless setUser has been called again with different data.